projects/dexfinance-vault/index.js

```diff
@@ -0,0 +1,156 @@
+    factory: "0xac297968C97EF5686c79640960D106f65C307a37",
+    USDEX_PLUS: "0xaA2C47a35C1298795B5271490971Ec4874C8e53d",
+  },
+  base: {
+    factory: "0x714c94b9820d7d73e61510e4c18b91f995a895c1",
+  },
+  arbitrum: {
+    factory: "0xe31fceaf93667365ce1e9edad3bed4a7dd0fc01a",
+    USDEX_PLUS: "0x4117EC0A779448872d3820f37bA2060Ae0B7C34B",
+    gDEX: "0x92a212d9f5eef0b262ac7d84aea64a0d0758b94f"
+  },
+  avax: {
+    factory: "0x6b714e6296b8b977e1d5ecb595197649e10a3db1",
+  },
+  bsc: {
+    factory: "0x3ace08b10b5c08a17d1c46277d65c81249e65f44",
+  },
+  // manta: {
+  //   factory: "0x714C94B9820D7D73e61510e4C18B91F995A895C1",
+  // },
+};
+
+const getVaults = async (api, factory) => {
+  const vaults = await api.fetchList({ lengthAbi: abi.factory.vaultsLength, itemAbi: abi.factory.vaults, target: factory, });
+  const farmsAll = await api.fetchList({ lengthAbi: abi.vault.farmsLength, itemAbi: abi.vault.farms, targets: vaults, groupedByInput: true })
+  const items = []
+  vaults.map((vault, i) => items.push(farmsAll[i].map((farm) => ({ vault, farm }))))
+  return items.flat()
+};
+
+const getVaultsConnectors = async (api, factory, vaultFarms) => {
+  const connectorsCalls = vaultFarms.map(({ farm, vault }) => ({ params: farm.beacon, target: vault, }));
+  const calculationConnectorCalls = vaultFarms.map(({ farm }) => farm.beacon);
+  const connectors = await api.multiCall({ abi: abi.vault.farmConnector, calls: connectorsCalls })
+  const calculationConnectors = await api.multiCall({ abi: abi.factory.farmCalculationConnector, calls: calculationConnectorCalls, target: factory })
+
+  vaultFarms.forEach((item, i) => {
+    delete item.farm.data
+    item.connector = connectors[i]
+    item.calculationConnector = calculationConnectors[i]
+  })
+}
+
+const getVaultsDatas = async (api, vaultFarms) => {
+  const calls = vaultFarms.map(({ connector }) => connector)
+  const liquidityCalls = vaultFarms.map(({ calculationConnector, connector }) => ({ target: calculationConnector, params: [connector] }))
+
+  const [types, stakingTokens, tokenIds, liquidities] =
+    await Promise.all([
+      api.multiCall({ calls, abi: abi.farm.type }),
+      api.multiCall({ calls, abi: abi.farm.stakingToken }),
+      api.multiCall({ calls, abi: abi.farm.tokenId, permitFailure: true }),
+      api.multiCall({ calls: liquidityCalls, abi: abi.vault.liquidity }),
+    ]);
+
+  vaultFarms.forEach((item, i) => {
+    item.type = types[i]
+    item.stakingToken = stakingTokens[i]
+    item.liquidity = liquidities[i]
+    item.tokenId = tokenIds[i]
+  })
+};
+
+const groupBy = (array, keyFn) => {
+  return array.reduce((acc, item) => {
+    const key = keyFn(item);
+    if (!acc[key]) {
+      acc[key] = [];
+    }
+    acc[key].push(item);
+    return acc;
+  }, {});
+};
+
+const lpv2Balances = async (api, farms) => {
+  farms.forEach(({ stakingToken, liquidity }) => {
+    api.add(stakingToken, liquidity);
+  });
+};
+
+const lpv3Balances = async (api, farms) => {
```

Comment:
why not use the code with positionIds? rest of the PR is good

Reply:
I'm using it right now, it works well :)